cmd/soroban-rpc/internal/db/db.go

```diff
@@ -128,33 +128,37 @@ func getMetaValue(ctx context.Context, q db.SessionInterface, key string) (strin
 	case 1:
 		// expected length on an initialized DB
 	default:
-		return "", fmt.Errorf("multiple entries (%d) for key %q in table %q", len(results), latestLedgerSequenceMetaKey, metaTableName)
+		return "", fmt.Errorf("multiple entries (%d) for key %q in table %q", len(results), key, metaTableName)
 	}
 	return results[0], nil
 }
 
-func getLatestLedgerSequence(ctx context.Context, q db.SessionInterface, cache *dbCache) (uint32, error) {
-	latestLedgerStr, err := getMetaValue(ctx, q, latestLedgerSequenceMetaKey)
-	if err != nil {
-		return 0, err
+func getLatestLedgerSequence(ctx context.Context, ledgerReader LedgerReader, cache *dbCache) (uint32, error) {
+	cache.RLock()
+	latestLedgerSeqCache := cache.latestLedgerSeq
+	cache.RUnlock()
+
+	if latestLedgerSeqCache != 0 {
+		return latestLedgerSeqCache, nil
 	}
```

Comment:
Are you sure this can be done now? 

I am not saying it can't. It's just that I haven't looked at this code for a while and from our latest conversation there were failing tests and races when enabling this. I would like to know what changed to make this possible.

Reply:
@2opremio The failing tests earlier were happening because I did not lock the cache when reading the value:
https://github.com/stellar/soroban-rpc/pull/256/files#diff-2b645f26f8eb502d1988698186482a5910b37509e606f968a3a60c927c397cfeR137

After adding the `Rlock()` function, there are no races encountered